server/src/test/validator.test.ts

```
@@ -66,3 +71,54 @@ suite('validateNamesAndMappings', () => {
 		assert.deepStrictEqual(actualDiagnostic.range, pinWithDuplicateName.range);
 	}); 
 });
+
+suite('validatePinBlock', () => {
+
+	// unmock the file system after each test
+	teardown(mockfs.restore);
+
+	test('Validate Conflict Based On Pin Block', () => {
+		const pins = [
+			{ name: 'MY_LED', type: 'Gpio', mapping: 'MT3620_GPIO4' },
+			{ name: 'MY_PWM_CONTROLLER0', type: 'Pwm', mapping: 'MT3620_PWM_CONTROLLER1' }
+		];
+		mockfs({
+			'my_app/odm.json':
+				`
+				{
+					"Metadata": { "Type": "Azure Sphere Hardware Definition", "Version": 1 },
+					"Imports": [ { "Path": "mt3620.json" } ],
+					"Peripherals": [
+						{ "Name": "${pins[0].name}", "Type": "${pins[0].type}", "Mapping": "${pins[0].mapping}" },
+						{ "Name": "${pins[1].name}", "Type": "${pins[1].type}", "Mapping": "${pins[1].mapping}" }
+					]
+				}
+				`,
+			'my_app/mt3620.json':
+				`
+				{
+					"Metadata": { "Type": "Azure Sphere Hardware Definition", "Version": 1 },
+					"Peripherals": [
+						{ "Name": "${pins[0].mapping}", "Type": "${pins[0].type}", "AppManifestValue": 4 },
+						{ "Name": "${pins[1].mapping}", "Type": "${pins[1].type}", "AppManifestValue": "PWM-CONTROLLER-1" }
+					]
+				}
+				`
+		});
+
+		const hwDefFilePath = 'my_app/odm.json';
+		const hwDefinition = tryParseHardwareDefinitionFile(fs.readFileSync(hwDefFilePath, { encoding: 'utf8' }), asURI(hwDefFilePath), '');
+
+		if (hwDefinition) {
```

Should fail test if hardware definition is undefined